packages/financial-templates-lib/test/price-feed/UniswapV3PriceFeed.js

```diff
@@ -1,22 +1,22 @@
+const { web3, getContract, network } = require("hardhat");
+const { assert } = require("chai");
 const { toWei, toBN } = web3.utils;
 const winston = require("winston");
 const bn = require("bignumber.js");
 
-const { UniswapV3PriceFeed } = require("../dist/price-feed/UniswapPriceFeed");
-const { mineTransactionsAtTime, MAX_SAFE_JS_INT, parseFixed } = require("@uma/common");
-const { delay } = require("../dist/helpers/delay.js");
-const { getTruffleContract } = require("@uma/core");
+const { UniswapV3PriceFeed } = require("../../dist/price-feed/UniswapPriceFeed");
+const { mineTransactionsAtTimeHardhat, MAX_SAFE_JS_INT, parseFixed } = require("@uma/common");
+const { delay } = require("../../dist/helpers/delay.js");
 
-const CONTRACT_VERSION = "latest";
-
-const UniswapMock = getTruffleContract("UniswapV3Mock", web3, CONTRACT_VERSION);
-const Uniswap = getTruffleContract("UniswapV3", web3, CONTRACT_VERSION);
-const Token = getTruffleContract("ExpandedERC20", web3, CONTRACT_VERSION);
+const UniswapMock = getContract("UniswapV3Mock");
+const Uniswap = getContract("UniswapV3");
+const Token = getContract("ExpandedERC20");
 
 const Convert = (decimals) => (number) => (number ? parseFixed(number.toString(), decimals).toString() : number);
+const getFutureTime = async () => Number((await web3.eth.getBlock("latest")).timestamp) + 10000;
```

Comment:
niiice. sneaky

Reply:
Yeah, these tests all do some very weird things. They were stuck running on oldschool truffle (no hardhat at all) because they didn't play nicely with the hardhat node. I realized that this was because 1) some of our RPC calls to move time around were not allowed/ignored by hardhat (which is why I made the change in common) and 2) we were actually mining blocks with timestamps that were before blocks that came before them. Ganache allows this, but hardhat does not. So this method ensures that each test starts pretty far past the previous one so it has space to mine pretty far before this timestamp without breaking.

Reply:
very nice. thanks for the context!